ca/certificates_test.go

```diff
@@ -741,3 +744,150 @@ func TestNewRootCAWithPassphrase(t *testing.T) {
 	assert.Contains(t, string(anotherNewRootCA.Key), "Proc-Type: 4,ENCRYPTED")
 
 }
+
+type certTestCase struct {
+	cert        []byte
+	errorStr    string
+	root        []byte
+	allowExpiry bool
+}
+
+func TestValidateCertificateChain(t *testing.T) {
+	leaf, intermediate, root := testutils.ECDSACertChain[0], testutils.ECDSACertChain[1], testutils.ECDSACertChain[2]
+	intermediateKey, rootKey := testutils.ECDSACertChainKeys[1], testutils.ECDSACertChainKeys[2] // we don't care about the leaf key
+
+	var chain func(certs ...[]byte) []byte
```

Comment:
It's recursive - I can't refer back to itself before it's declared otherwise.  I can just change this to an iterative function (I only did it recursively to see if I could declare a recursive function using `chain :=`)